models/article.models.js

```diff
@@ -24,9 +24,12 @@ const fetchArticleById = (article_id) => {
     }
 
     const selectArticleById = `
-        SELECT * 
-        FROM articles 
-        WHERE article_id = $1;
+        SELECT A.*,  COUNT(C.article_id) AS comment_count 
```

Comment:
I really like this casting here, it makes your sql look very tidy, there is a chance it could make it less readable but again totally fine here :)